services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParser.java

```diff
@@ -320,13 +320,10 @@ public VenicePath parseResourceUri(String uri, BasicFullHttpRequest fullHttpRequ
       }
 
       AggRouterHttpRequestStats aggRouterHttpRequestStats = routerStats.getStatsByType(requestType);
-      if (!requestType.equals(SINGLE_GET)) {
-        /**
-         * Here we only track key num for non single-get request, since single-get request will be always 1.
-         */
-        keyNum = path.getPartitionKeys().size();
-        aggRouterHttpRequestStats.recordKeyNum(storeName, keyNum);
-      }
+
+      // record key num details for all types of requests to keep the metrics behavior uniform
+      keyNum = path.getPartitionKeys().size();
+      aggRouterHttpRequestStats.recordKeyNum(storeName, keyNum);
```

Comment:
This looks maybe a touch heavy to me?  I don't think we need to address it in this PR, but walking down this code path we're gonna generate some objects for every single key lookup.  We should monitor GC behavior/latency when enabling otel after this change.

Reply:
If we have a safety measure coming up I think we should either throw the above tweak behind a config that we remove once your cache changes are ready, or, we just don't include this portion in this PR?  My fear is we go to the site with something that is detrimental to performance.

Reply:
Agree and reverted this change in this PR. Will add it as part of the next PR wiht the safety measure. Thanks.